MatrixInterpolator.h

```diff
@@ -0,0 +1,141 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: Computes the MatrixInterpolator algorithm on the given snapshot matrix.
+
+#ifndef included_MatrixInterpolator_h
+#define included_MatrixInterpolator_h
+
+#include "Interpolator.h"
+#include <vector>
+#include <string>
+
+namespace CAROM {
+
+class Matrix;
+class Vector;
+
+/**
+ * MatrixInterpolator interpolates reduced matrices of a set of parameter points
+ * and returns an interpolated reduced matrix for an unseen parameter point.
+ * The interpolation algorithm was adapted from "Gradient-based
+ * Constrained Optimization Using a Database of Linear Reduced-Order Models"
+ * by Y. Choi et al.
+ */
+class MatrixInterpolator : public Interpolator
+{
+public:
+
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] parameter_points The parameter points.
+     * @param[in] rotation_matrices The rotation matrices associated with
+     *                              each parameter point.
+     * @param[in] reduced_matrices The reduced matrices associated with
+     *                             each parameter point.
+     * @param[in] ref_point The index within the vector of parameter points
+     *                      to the reference point
+     * @param[in] matrix_type The type of matrix (R = real, B = basis,
```

Comment:
It's not clear what `R` and `B` are. There are 3 methods of interpolation, `NS`, `SPD`, and general (Y-X), so I don't see how these 4 matrix types correspond to the 3 methods. More documentation on the matrix and interpolation types would be helpful.

Reply:
R = general (Y-X)
B = basis. For basis, doing the rotation is a bit different, then with general Y-X. I'll add more documentation